arch/x86/boot/compressed/sl_main.c

```diff
@@ -147,9 +147,20 @@ static void sl_find_event_log(struct tpm *tpm)
 
 	if (sl_cpu_type == SL_CPU_AMD) {
 		struct sl_header *sl_hdr = sl_lz_base;
-		struct lz_header *lz_hdr = sl_lz_base + sl_hdr->lz_length;
+		struct lz_tag_tags_size *t = sl_lz_base + sl_hdr->bootloader_data_offset;
+		struct lz_tag_evtlog *t_log = (struct lz_tag_evtlog *)t;
+		void *end = (void *)t + t->size;
+
+		while ((void *)t_log < end
+		       && t_log->hdr.type != LZ_TAG_EVENT_LOG
+		       && t_log->hdr.type != LZ_TAG_END) {
+			t_log = (void *)t_log + t_log->hdr.len;
+		}
+
+		if (t_log->hdr.type != LZ_TAG_EVENT_LOG)
+			return;
```

Comment:
Should this be immediate death if there is no logging structure? Not much else is gonna work.

Reply:
Probably yes. This is leftover from when the code changed frequently to have some compatibility with older SL implementations in bootloaders. Now GRUB should fill it even when there is no ACPI table so it should work anyway.

Reply:
ack